File: 7 kyu/Doubleton number/solution.rb
def doubleton(num)
  num.succ.step.find{ |i| i.digits.uniq.size == 2 }
end

#############
def doubleton(num)
  for i in (num+1)..1000000
    if i.to_s.chars.sort.join.squeeze.size == 2
      return i 
    end
  end
end

###################
def doubleton(num)
  (num+1..1000000).each do |num|
    return num if num.digits.uniq.size == 2
  end
end

##############
def doubleton(num)
  loop do
    num += 1
    return num if num.digits.uniq.size == 2
  end
end

################
def doubleton(num)
  answer = num + 1
  answer += 1 while answer.digits.uniq.count != 2
  return answer
end
  
#################
def doubleton(num)
  until (num += 1).digits.group_by(&:itself).keys.count == 2; end; num
end
  
##########
def doubleton(num)
    n = num + 1
    while ((n.to_s).split("").uniq).length != 2
        n += 1
    end
    return n
end

File: 7 kyu/Doubleton number/solution.jl
function doubleton(num::Integer)::Int
  while true
    num += 1
    length(Set(digits(num))) == 2 && return num
  end
end

##############
function doubleton(num::Integer)::Int
  num += 1
  while length(unique(digits(num))) != 2
    num += 1
  end
  num  
end

############
function doubleton(num::Integer)::Int
    while true
        num += 1
        if length(Set(digits(num)))==2
            return num
        end
    end
end

################
function doubleton(num::Integer)::Int
  num += 1
  while length(unique(digits(num))) ≠ 2
    num += 1
  end
  num
end

###############
function doubleton(num::Integer)::Int
  length(unique(digits(num+1))) == 2 ? num+1 : doubleton(num+1)
end
